### PR TITLE
Support using 'stack --docker'

### PR DESCRIPTION
**Pull request by jcberentsen**
Tested on OSX

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,29 @@ Currently the component that takes the longest time to compile is
...
 
+## Stack development with Docker
+This is an alternative way to use Docker than above, taking advantage of stack's Docker support.
+Makes it easy to install new packages with `stack --docker install <pkg>` without having to rebuild a docker image.
+Like the other Docker workflow, this doesn't require any host dependecies to be installed.
+
+```
+docker build -t ihaskell-dev docker
+stack --docker setup
+stacl --docker install
```
Comment (vaibhavsagar) [inline README.md]: Please fix this typo.

File: README.md [hunk 2]
```diff
@@ -81,6 +81,29 @@ Currently the component that takes the longest time to compile is
...
+stack --docker exec ihaskell -- install --stack
+stack --docker exec jupyter -- notebook --ip=0.0.0.0 notebooks
+```
+
+Everything in the LTS should be available!
```
Comment (vaibhavsagar) [inline README.md]: I don't see how this is true. Does the `Dockerfile` somehow download and install all packages in the LTS? If not, please reword this to say something like "Everything in the LTS can be made available!"

File: README.md [hunk 3]
```diff
@@ -81,6 +81,29 @@ Currently the component that takes the longest time to compile is
...
+To add a a package outside the LTS, simpy add it to the `stack.yaml` file  (See: "Where are my packages?" below).
```
Comment (vaibhavsagar) [inline README.md]: Following on from my comment above, please change this to "To add a package"

File: README.md [hunk 4]
```diff
@@ -81,6 +81,29 @@ Currently the component that takes the longest time to compile is
...
+Then install the package with stack before restarting `jupyter`
+
+```
+# add details about mypackage to stack.yaml
```
Comment (vaibhavsagar) [inline README.md]: Can this be reworded to something like this?

```
# after adding details about mypackage to stack.yaml
```

File: README.md [hunk 5]
```diff
@@ -81,6 +81,29 @@ Currently the component that takes the longest time to compile is
...
+stack --docker install mypackage
+stack --docker exec jupyter -- notebook notebooks
+```
+
```
Comment (vaibhavsagar) [inline README.md]: One newline instead of two, please.

File: README.md [hunk 6]
```diff
@@ -81,6 +81,29 @@ Currently the component that takes the longest time to compile is
...
+This is an alternative way to use Docker than above, taking advantage of stack's Docker support.
+Makes it easy to install new packages with `stack --docker install <pkg>` without having to rebuild a docker image.
+Like the other Docker workflow, this doesn't require any host dependecies to be installed.
+
+```
+docker build -t ihaskell-dev docker
+stack --docker setup
+stacl --docker install
+stack --docker exec ihaskell -- install --stack
+stack --docker exec jupyter -- notebook --ip=0.0.0.0 notebooks
```
Comment (vaibhavsagar) [inline README.md]: Can we move some of these instructions out of the `README` and into the `Dockerfile`? I think everything up until the final `stack --docker exec jupyter -- notebook --ip=0.0.0.0 notebooks` should be safe to put in the `Dockerfile`. If not, why?

**Comment by vaibhavsagar** [comment]
Thanks so much for this contribution! I'm generally happy with this and looking forward to merging it eventually, but I have a few concerns/questions first.

File: README.md (hunk 3 above)
Comment (vaibhavsagar) [reply README.md]: Also s/simpy/simply/.

File: README.md (hunk 2 above)
Comment (jcberentsen) [reply README.md]: Agree, "can be made available" is more accurate

File: README.md (hunk 6 above)
Comment (jcberentsen) [reply README.md]: The docker image used doesn't actually contain the IHaskell source, so `stack` can't see the `stack.yaml` which is outside the docker build context, can it?

All the build artifacts and packages will be on the _host_ (volume mounted into the docker container), which allows installing additional packages without rebuilding the docker image and keeping the docker container ephemeral.

Comment (vaibhavsagar) [reply README.md]: You're right, I had forgotten about this.

**Comment by vaibhavsagar** [comment]
Thanks again!
